tests/components/nintendo_parental/conftest.py

```diff
@@ -0,0 +1,119 @@
+"""Common fixtures for the Nintendo Switch Parental Controls tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from pynintendoparental import Device
+import pytest
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.nintendo_parental.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def mock_nintendo_client_init() -> Generator[MagicMock]:
+    """Return a mocked NintendoParental."""
+    with (
+        patch(
+            "homeassistant.components.nintendo_parental.config_flow.NintendoParental",
+            autospec=True,
+        ) as nintendo_parental_client,
+        patch(
+            "homeassistant.components.nintendo_parental.NintendoParental",
+            new=nintendo_parental_client,
+        ),
+    ):
+        yield nintendo_parental_client
+
+
+@pytest.fixture
+def mock_request_handler() -> Generator[MagicMock]:
+    """Mock the _request_handler function."""
+    with patch(
+        "pynintendoparental.authenticator.Authenticator._request_handler",
```

Comment:
1. we should patch this where we use it
2. Why to we patch itnernals?

Reply:
1. good question
2. good question

Wrote these tests with duck tape over my eyes ;) but I've changed a fair bit in the Python module itself.... All of my py modules hid all of the HTTP methods behind something really generic, at the time it was a case of "adding a new api endpoint will be easy this way"... a couple of years later its proved to be a nightmare when debugging.

I will rewrite these.